drivers/i2c/i2c_ll_stm32.h

```diff
@@ -91,6 +104,14 @@ struct i2c_stm32_data {
 	i2c_stm32_smbalert_cb_func_t smbalert_cb_func;
 	const struct device *smbalert_cb_dev;
 #endif
+#ifdef I2C_STM32_V2_DMA
+	struct dma_config dma_tx_cfg;
+	struct dma_block_config dma_tx_blk_cfg;
+	struct k_sem dma_tx_sem;
+	struct dma_config dma_rx_cfg;
+	struct dma_block_config dma_rx_blk_cfg;
+	struct k_sem dma_rx_sem;
```

Comment:
As I2C is half-duplex only, Structs dma_config, dma_block_config could share the same instance for both rx and tx
Would save some RAM

Reply:
Hold over from implementing separate tx and rx routines. I will look to use a single config block

Reply:
Converted to using single blk and config struct for dma